### ISSUE TITLE
Simplify reading strings

### ISSUE THREAD

**Issue by matthew-d-jones** [ess-dmsc/h5cpp#317]
Reading a scalar variable length string is simple, this works:
```cpp
auto attr = aNode.attributes["an_attribute"];
std::string anAttribute;
attr.read(anAttribute);
```
However, if the string happens to be fixed length then the above code fails with an unhelpful `std::bad_alloc` thrown.
The following (for which there is no example) does work for the fixed string:
```cpp
auto attr = aNode.attributes["an_attribute"];
auto readType = hdf5::datatype::String::fixed(attr.datatype().size());
std::string anAttribute;
attr.read(anAttribute, readType);
```

I don't think I should have to care, for reading, whether the string is fixed or variable length. Ideally the first code snippet should work in both cases.

**Comment by matthew-d-jones** [comment]
@jkotan More string problems, but it's a slightly different situation and the fix in #309 of just doing: 
```
attr.read(testattr,attr.datatype());
```
does not work in this case. I actually had to create a new datatype object and set the size on it.

**Comment by matthew-d-jones** [comment]
I just checked again, the above does work in this case. I'll close this ticket and add to discussion in #309.
